src/zks.rs

```diff
@@ -241,13 +265,49 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> ZksNamespaceT
         }
     }
 
+    /// Get all known balances for a given account.
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - The user address with balances to check.
+    ///
+    /// # Returns
+    ///
+    /// A `BoxFuture` containing a `Result` with a (Token, Balance) map where account has non-zero value.
     fn get_all_account_balances(
         &self,
-        _address: zksync_basic_types::Address,
+        address: zksync_basic_types::Address,
     ) -> jsonrpc_core::BoxFuture<
         jsonrpc_core::Result<std::collections::HashMap<zksync_basic_types::Address, U256>>,
     > {
-        not_implemented("zks_getAllAccountBalances")
+        let inner = self.node.clone();
+        Box::pin({
+            let address = address.clone();
+            self.get_confirmed_tokens(0, 100)
+                .then(move |tokens| async move {
+                    let tokens =
+                        tokens.map_err(|_err| into_jsrpc_error(Web3Error::InternalError))?;
+
+                    let mut writer = inner
+                        .write()
```

Comment:
Could this be a `.read()`? I don't think we're updating any values within this API.

Reply:
i don't think so- we may mutate the cache- `fork_storage.read_value()` takes `&mut self`